src/pycram/worlds/multiverse_communication/socket.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+
+"""Multiverse Client base class."""
+
+import rospy
+from multiverse_client_pybind import MultiverseClientPybind  # noqa
+from typing_extensions import Optional, List, Dict, Callable, TypeVar
+
+from ..multiverse_datastructures.dataclasses import MultiverseMetaData
+from ...config import multiverse_conf as conf
+
+T = TypeVar("T")
+
+
+class MultiverseSocket:
+
+    def __init__(
+            self,
+            port: str,
+            host: str = conf.HOST,
+            meta_data: MultiverseMetaData = MultiverseMetaData(),
+    ) -> None:
+        """
+        Initialize the MultiverseSocket, connect to the Multiverse Server and start the communication.
+
+        :param port: The port of the client.
+        :param host: The host of the client.
+        :param meta_data: The metadata for the Multiverse Client as MultiverseMetaData.
+        """
+        if not isinstance(port, str) or port == "":
+            raise ValueError(f"Must specify client port for {self.__class__.__name__}")
+        self._send_data = None
+        self.port = port
+        self.host = host
+        self._meta_data = meta_data
+        self.client_name = self._meta_data.simulation_name
+        self._multiverse_socket = MultiverseClientPybind(
+            f"{conf.SERVER_HOST}:{conf.SERVER_PORT}"
+        )
+        self.request_meta_data = {
+            "meta_data": self._meta_data.__dict__,
+            "send": {},
+            "receive": {},
+        }
+        self._api_callbacks: Optional[Dict] = None
+
+        self._start_time = 0.0
+
+    def run(self) -> None:
+        """Run the client."""
+        self.log_info("Start")
```

Comment:
Log more than just "Start" Like "start multiverse socket"

Reply:
Yes, the log_info method adds a message template to each given message, this template adds the class name (because multiverse_clients inherit from this class, the class name will be different depending on the client class name, for e.g. MultiverseReader, MultiverseWriter, ...etc.), the port used for the socket, the sim time and world time.